specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/favorites_API.json

```diff
@@ -0,0 +1,403 @@
+        },
+        "FavoriteId": {
+          "type": "string",
+          "description": "Internally assigned unique id of the favorite definition.",
+          "readOnly": true
+        },
+        "FavoriteType": {
+          "description": "Enum indicating if this favorite definition is owned by a specific user or is shared between all users with access to the Application Insights component.",
+          "type": "string",
+          "enum": [
+            "shared",
+            "user"
+          ],
+          "x-ms-enum": {
+            "name": "FavoriteType",
+            "modelAsString": false
+          }
+        },
+        "SourceType": {
+          "type": "string",
+          "description": "The source of the favorite definition."
+        },
+        "TimeModified": {
+          "type": "string",
+          "description": "Date and time in UTC of the last modification that was made to this favorite definition.",
+          "readOnly": true
+        },
+        "Tags": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of 0 or more tags that are associated with this favorite definition"
+        },
+        "Category": {
+          "type": "string",
+          "description": "Favorite category, as defined by the user at creation time."
+        },
+        "IsGeneratedFromTemplate": {
+          "type": "boolean",
+          "description": "Flag denoting wether or not this favorite was generated from a template."
+        },
+        "UserId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Unique user id of the specific user that owns this favorite."
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The Azure subscription ID."
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Client Api Version."
+    },
+    "ResourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the resource group.",
+      "x-ms-parameter-location": "method"
+    },
+    "ResourceNameParameter": {
+      "name": "resourceName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the Application Insights component resource.",
+      "x-ms-parameter-location": "method"
+    },
+    "FavoriteTypeParameter": {
+      "name": "favoriteType",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "default": "shared",
+      "enum": [
+        "shared",
+        "user"
+      ],
+      "description": "The type of favorite. Value can be either shared or user.",
```

Comment:
`x-ms-enum` missing, also please don't duplicate the `enum` information in the description 🙂

Reply:
Only SourceTypeParameter needs to be modelled as a string, not FavoriteTypeParameter (sorry for any confusion on past comments).

Also, I added the `x-ms-enum` field in the missing instance.

Reply:
no worries, thanks for clarifying 🙂